relay-general/src/store/normalize/span/tag_extraction.rs

```diff
@@ -0,0 +1,693 @@
+/// truncates it to the maximum allowed size and sets the last character to
+/// `*`.
+fn truncate_string(mut string: String, max_bytes: usize) -> String {
+    if string.len() <= max_bytes {
+        return string;
+    }
+
+    if max_bytes == 0 {
+        return String::new();
+    }
+
+    let mut cutoff = max_bytes - 1; // Leave space for `*`
+
+    while cutoff > 0 && !string.is_char_boundary(cutoff) {
+        cutoff -= 1;
+    }
+
+    string.truncate(cutoff);
+    string.push('*');
+    string
+}
+
+/// Regex with a capture group to extract the database action from a query.
+///
+/// Currently, we're only interested in `SELECT`, `INSERT`, `DELETE` and `UPDATE` statements.
+static SQL_ACTION_EXTRACTOR_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r#"(?i)(?P<action>(SELECT|INSERT|DELETE|UPDATE))"#).unwrap());
+
+fn sql_action_from_query(query: &str) -> Option<&str> {
+    extract_captured_substring(query, &SQL_ACTION_EXTRACTOR_REGEX)
+}
+
+/// Regex with a capture group to extract the table from a database query,
+/// based on `FROM` and `INTO` keywords.
+static SQL_TABLE_EXTRACTOR_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r#"(?i)(from|into)(\s|"|'|\()+(?P<table>(\w+(\.\w+)*))(\s|"|'|\))+"#).unwrap()
+});
+
+/// Returns the table in the SQL query, if any.
+///
+/// If multiple tables exist, only the first one is returned.
+fn sql_table_from_query(query: &str) -> Option<&str> {
+    extract_captured_substring(query, &SQL_TABLE_EXTRACTOR_REGEX)
+}
+
+/// Regex with a capture group to extract the HTTP method from a string.
+static HTTP_METHOD_EXTRACTOR_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r#"(?i)^(?P<method>(GET|HEAD|POST|PUT|DELETE|CONNECT|OPTIONS|TRACE|PATCH))\s"#)
+        .unwrap()
+});
+
+fn http_method_from_transaction_name(name: &str) -> Option<&str> {
+    extract_captured_substring(name, &HTTP_METHOD_EXTRACTOR_REGEX)
+}
+
+/// Returns the captured substring in `string` with the capture group in `pattern`.
+///
+/// It assumes there's only one capture group in `pattern`, and only returns the first one.
+fn extract_captured_substring<'a>(string: &'a str, pattern: &'a Lazy<Regex>) -> Option<&'a str> {
+    let capture_names: Vec<_> = pattern.capture_names().flatten().collect();
+
+    for captures in pattern.captures_iter(string) {
+        for name in &capture_names {
+            if let Some(capture) = captures.name(name) {
+                return Some(&string[capture.start()..capture.end()]);
+            }
+        }
+    }
+
+    None
+}
+
+/// Returns the category of a span from its operation. The mapping is available in:
+/// <https://develop.sentry.dev/sdk/performance/span-operations/>
+fn span_op_to_category(op: &str) -> Option<&str> {
+    let category =
+        // General
+        if op.starts_with("mark") {
+            "mark"
+        }
+        //
+        // Browser
+        // `ui*` mapped in JS frameworks
+        else if op.starts_with("pageload") {
```

Comment:
_nit_: As per convention, our span ops are always separated with a `'.'`. This means you can call split_once and then match on it. 

I'm pretty sure we have similar code somewhere else already, probably in breakdown computation.

Reply:
Will do this in a follow-up.